### PR TITLE
Add mongoDB plugin report db instance info

### PR DESCRIPTION
**Pull request by hxd123456**
If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes https://github.com/apache/skywalking/issues/11252.
 Update the [CHANGES log](https://github.com/apache/skywalking-java/blob/main/CHANGES.md).

### CODE REVIEW

**Comment by wu-sheng** [comment]
Please format the PR, this seems no clear description.

**Comment by hxd123456** [comment]
> Please format the PR, this seems no clear description.

I have enriched the description information, please check the PR

File: apm-sniffer/apm-sdk-plugin/mongodb-4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/mongodb/v4/support/MongoSpanHelper.java [hunk 1]
```diff
@@ -43,6 +44,13 @@ public static void createExitSpan(String executeMethod, String remotePeer, Objec
         Tags.DB_TYPE.set(span, MongoConstants.DB_TYPE);
         SpanLayer.asDB(span);
 
+        if (operation instanceof EnhancedInstance) {
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/mongodb-4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/mongodb/v4/support/MongoSpanHelper.java:47]: What is the case not `instanceof`? Or this is just an extra check?

Comment (hxd123456) [reply apm-sniffer/apm-sdk-plugin/mongodb-4.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/mongodb/v4/support/MongoSpanHelper.java:47]: For example,'CommitTransactionOperation' cannot obtain the corresponding database info

File: CHANGES.md [hunk 2]
```diff
@@ -12,7 +12,7 @@ Release Notes.
 * Support customization for the length limitation of string configurations
 * Add max length configurations in `agent.config` file for service_name and instance_name
 * Optimize spring-cloud-gateway 2.1.x, 3.x witness class.
-
+* Support report mongodb instance info in mongodb 4.x plugin
```
Comment (wu-sheng) [inline CHANGES.md:15]:
```suggestion
* Support report MongoDB instance info in Mongodb 4.x plugin.

```

**Review by wu-sheng — APPROVED** [review]
LGTM

**Comment by wu-sheng** [comment]
For others reading this PR, the context happens at https://github.com/apache/skywalking-java/pull/597

**Pull request merged**